sync.py

```diff
@@ -96,22 +117,45 @@ def composed_modules2tagged_builds(composed):
         ret.append(ent)
     return ret
 
+def check_denylist_builds(builds, denylist):
+    """
+    Look for any builds on the denylist, and remove them.
+    """
+    ret = []
+    for build in builds:
+        if build['package_name'] in denylist:
+            continue
+        ret.append(build)
+    return ret
+
 def check_unsynced_builds(tagged_builds, packages_to_track):
     """
     Look for builds that are not synced with centos streams
     """
     unsynced_builds = []
 
-    for build in tagged_builds:
+    corootdir = "/tmp/centos-sync-" + str(os.getpid()) + "/"
+    print("Using tmp dir:", corootdir)
+    os.makedirs(corootdir)
+    for build in sorted(tagged_builds, key=lambda x: x['package_name']):
         if build['package_name'] in packages_to_track:
-            os.system("rm -rf /tmp/" + build['package_name'])
-            repo = git.Repo.clone_from("https://git.centos.org/rpms/" + build['package_name'] + ".git", "/tmp/" +
-            build['package_name'])
-           
-            tag_to_check = "imports/c8s/" + build['nvr']
+            codir = corootdir + build['package_name']
+            os.system("rm -rf " + codir)
+
+            giturl = "https://git.centos.org/rpms/"
+            giturl += build['package_name']
+            giturl += ".git"
+            try:
+                repo = git.Repo.clone_from(giturl, codir)
+                tags = repo.tags
+            except git.exc.GitCommandError:
+                # This means the clone didn't work, so it's a new package.
+                tags = []
+
+            tags_to_check = ("imports/c8s/" + build['nvr'], "imports/c8/" + build['nvr'])
             new_build = True
-            for tag in repo.tags:
-                if tag_to_check ==  str(tag):
+            for tag in tags:
+                if str(tag) in tags_to_check:
```

Comment:
If this check is successful, it would be worth logging what tag actually passed this check. It would be helpful for debugging in future.

Reply:
Doesn't line 160 do that?

Reply:
Sorry, I misunderstood that `print` statement. You are correct.